homework/index.js

```diff
@@ -0,0 +1,206 @@
+class ShoppingMarketError extends Error {
+    static ERROR_CODE = {
+        EXPIRED_ALIMENT: 'EXPIRED_ALIMENT',
+        NOT_IN_CART: 'NOT_IN_CART',
+        INSUFFICIENT_FUNDS: 'INSUFFICIENT_FUNDS'
+    };
+
+    static ERROR_MESSAGE = {
+        [ShoppingMarketError.ERROR_CODE.EXPIRED_ALIMENT]:
+            'Aliment is expired',
+
+        [ShoppingMarketError.ERROR_CODE.NOT_IN_CART]:
+            'Product is not in the shopping cart',
+
+        [ShoppingMarketError.ERROR_CODE.INSUFFICIENT_FUNDS]:
+            'Customer has insufficient funds'
+    };
+
+    constructor({
+        code,
+        message = ShoppingMarketError.ERROR_MESSAGE[code],
+        data
+    }) {
+        super(message);
+
+        this.code = code;
+        this.data = data;
+    }
+
+    toString() {
+        return JSON.stringify({
+            code: this.code,
+            message: this.message,
+            data: this.data
+        }, null, 4) + '\n' + this.stack;
+    }
+}
+
+class Person {
+    #name;
+
+    constructor(name) {
+        this.#name = name;
+    }
+
+    get name() {
+        return this.#name;
+    }
+}
+
+class Customer extends Person {
+    #money;
+    #shoppingCart;
+
+    constructor({ name = 'unknown', money }) {
+        super(name);
+
+        this.#money = money;
+        this.#shoppingCart = [];
+    }
+
+    get money() {
+        return this.#money;
+    }
+
+    get shoppingCart() {
+        return this.#shoppingCart;
```

Comment:
It's not a good idea to return a private reference. It makes the whole concept of `private` pointless since whoever use this getter gets access to the internal state of the Customer instance and can mess with it.